rag/llm/chat_model.py

```
@@ -2101,6 +2101,25 @@ async def _exec_tool(tc):
+
+                    # Terminal-tool short-circuit: a terminal tool (e.g. ``rag``)
+                    # already composes the final answer itself, so return its
+                    # result directly instead of feeding it back for another LLM
+                    # round. This mirrors the streaming path
+                    # (``async_chat_streamly_with_tools``), which otherwise lets
+                    # the outer model re-invoke ``rag`` on partial answers until
+                    # ``max_rounds`` is exhausted — the source of multi-hop
+                    # timeouts (single request exceeding the client budget).
+                    _terminal = getattr(self, "terminal_tools", None)
+                    if _terminal:
+                        for tc, name, args, result, err in results:
```

Ruff (0.16.1)

[warning] Loop control variable `tc` not used within loop body

Rename unused `tc` to `_tc`

(B007)